src/components/BrowserFilter/BrowserFilter.react.js

```diff
@@ -137,10 +161,18 @@ export default class BrowserFilter extends React.Component {
                 <Button
                   color="white"
                   primary={true}
-                  value="Apply these filters"
-                  width="245px"
+                  value={this.state.confirmName ? 'Confirm' : 'Save filters'}
+                  width="120px"
+                  onClick={() => this.state.confirmName ? this.save() : this.setState({confirmName: true})}
+                />
+                <Button
+                  color="white"
+                  primary={true}
+                  value="Apply filters"
```

Comment:
Can you remove the `filters` from all the buttons in the dialog? It's the filters dialog, so it's implicit that `Apply` means `Apply the filter".